wsclient/src/main/java/org/hpccsystems/ws/client/antlr/EclReader.java

```diff
@@ -0,0 +1,437 @@
+// Generated from ecl.g4 by ANTLR 4.5
+package org.hpccsystems.ws.client.antlr;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.antlr.v4.runtime.tree.ParseTreeProperty;
+import org.hpccsystems.ws.client.platform.DFUDataColumnInfo;
+import org.hpccsystems.ws.client.platform.DFURecordDefInfo;
+import org.hpccsystems.ws.client.platform.EclInfo;
+
+/**
+ * This class reads ECL and converts it to an EclInfo object. Currently it processes RECORD definitions only.
+ */
+public class EclReader extends EclBaseListener
```

Comment:
Same as the ecl grammar name; I thought keeping it generic would make it more flexible later on, but I can change it if you wish